ckan/views/user.py

```diff
@@ -656,20 +656,20 @@ def post(self):
                 # user, as that would reveal the existence of accounts with
                 # this email address)
                 for user_dict in user_list:
-                    # This is ugly, but we need the user object for the mailer,
-                    # and user_list does not return them
-                    logic.get_action(u'user_show')(
-                        context, {u'id': user_dict[u'id']})
-                    user_objs.append(context[u'user_obj'])
+                    # `user_list` returned the users,
+                    # so we know they exist here.
+                    user_objs.append(model.User.get(user_dict['id']))
 
         else:
             # Search by user name
             # (this is helpful as an option for a user who has multiple
             # accounts with the same email address and they want to be
             # specific)
             try:
-                logic.get_action(u'user_show')(context, {u'id': id})
-                user_objs.append(context[u'user_obj'])
+                # do `user_show` to check for user existing
+                user_dict = logic.get_action(u'user_show')(context,
```

Comment:
we just called user_list, what does calling user_show on each user give us?

Reply:
The `user_list` call was in the above condition, checking for emails with `if '@' in id`.

This part is for usernames. Does calling `model.User.get()` with a bogus username raise an error, or just come back empty? If it just comes back empty, then I will do:

 ```
user_obj = model.User.get()
if user_obj:
    user_objs.append(user_ibj)
 ```

Reply:
right, `else:` statements are important :-)

Yes you will get back `None` from User.get(x) when x isn't found.

Reply:
@wardi okay cool, I have super simplified it now